kernel/queue.c

```diff
@@ -84,13 +84,15 @@ static void prepare_thread_to_run(struct k_thread *thread, void *data)
 	z_ready_thread(thread);
 }
 
-static inline void handle_poll_events(struct k_queue *queue, uint32_t state)
+static inline bool handle_poll_events(struct k_queue *queue, uint32_t state)
 {
 #ifdef CONFIG_POLL
-	z_handle_obj_poll_events(&queue->poll_events, state);
+	return z_handle_obj_poll_events(&queue->poll_events, state);
 #else
 	ARG_UNUSED(queue);
 	ARG_UNUSED(state);
+
+	return false;
 #endif /* CONFIG_POLL */
```

Comment:
This block could also use IS_ENABLED

Reply:
Unfortunately, the poll_events field only exists when CONFIG_POLL is defined, so using IS_ENABLED() would lead to a build error.